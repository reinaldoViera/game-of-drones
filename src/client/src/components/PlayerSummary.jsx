import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Paper, Typography, withStyles, CircularProgress } from '@material-ui/core';
import { Query } from "react-apollo";
import gql from 'graphql-tag';

export const playerSummary = gql`
query playerSummary($id: ID!){
  playerSummary(id: $id) {
    matchs
    wins
  }
}
`

const styles = theme => ({
  root: {
    ...theme.mixins.gutters(),
    paddingTop: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit * 2,
  },
});

export class PlayerSummary extends Component {
  static propTypes = {
    name: PropTypes.string,
    id: PropTypes.any,
  }
  
  render() {
    return (
      <Query query={playerSummary} variables={{ id: this.props.id}}  pollInterval={3000}>
        {
          ({ loading, error, data }) => (
            <Paper className={this.props.classes.root} elevation={5}>
              <Typography variant="h5" component="h3">
                {
                  this.props.name
                }
              </Typography>
              {
                loading ? <CircularProgress /> :
                  <Typography variant="h5" component="h4">
                    {
                      `Matchs: ${data.playerSummary.matchs}`
                    }
                    <br/>
                    {
                      `Wins: ${data.playerSummary.wins}`
                    }
                  </Typography>
              }
            </Paper>
          )
        }

      </Query>
    )
  }
}

export default withStyles(styles)(PlayerSummary);
