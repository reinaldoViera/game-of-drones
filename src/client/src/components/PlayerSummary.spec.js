import {
    MockedProvider
} from 'react-apollo/test-utils';
import React from 'react';
import {PlayerSummary, playerSummary} from './PlayerSummary';
import renderer from 'react-test-renderer';
import wait from 'waait';


const mocks = [{
    request: {
        query: playerSummary(1)
    },
    result: {
        data: {
            playerSummary: {
                matchs: 5,
                wins: 3
            },
        },
    },
}]

describe('PlayerSummary component', () => {
    test('should render result data', async () => {
        const result = renderer.create( 
            <MockedProvider mocks={mocks} addTypename={false}>
                <PlayerSummary name="Player 1" id='1' classes={{}}/>
            </MockedProvider>,
        ).toJSON();
        await wait(1); // wait for response
        expect(result).toMatchSnapshot();
    })

})