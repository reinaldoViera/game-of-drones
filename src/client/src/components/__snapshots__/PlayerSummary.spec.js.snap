// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerSummary component should render result data 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-elevation5-8 MuiPaper-rounded-2"
>
  <h3
    className="MuiTypography-root-28 MuiTypography-h5-44"
  >
    Player 1
  </h3>
  <h4
    className="MuiTypography-root-28 MuiTypography-h5-44"
  >
    Matchs: 5
    <br />
    Wins: 3
  </h4>
</div>
`;
